src/compas_occ/geometry/curves/nurbs.py

```
@@ -493,3 +494,49 @@ def segmented(self, u, v, precision=1e-3):
         copy = self.copy()
         copy.segment(u, v, precision)
         return copy
+
+    def join(self, curve, precision=1e-4):
+        """Modifies this curve by joining it with another curve.
+
+        Parameters
+        ----------
+        curve : :class:`OCCNurbsCurve`
+            The curve to join.
+
+        precision : float, optional
+            Tolerance for continuity and multiplicity.
+
+        Returns
+        -------
+        None
+
+        """
+        converter = GeomConvert_CompCurveToBSplineCurve(self.occ_curve)
+        success = converter.Add(curve.occ_curve, precision)
+        if success:
+            self.occ_curve = converter.BSplineCurve()
```

not sure if we should let this fail silently, but i guess we will find out what the right approach is by using it...